### PR TITLE
Leaderboard Loading Time

### LINKED ISSUE #1482: Leaderboard is making dozens of graphql requests
### What happened?

The leaderboard takes a very long time to load. This is probably because it is making a separate graphql request for every (visible?) player:
```
guild_player(where: { playerId: { _eq: $playerId } }) {
   ...
}
```

image

### What did you expect to happen?

The leaderboard page loads reasonably quickly

### How can we reproduce the problem (as minimally as possible)?

Go to the leaderboard and open the developer tools > network tab

### Is there anything else we need to know?

_No response_

### Browser and operating system used (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by dysbulic**
## Overview

**What features/fixes does this PR include?**

The primary point of this PR is to move the loading of DAOHaus DAOs & Discord roles *(which require `fetch`ing external resources)* to a separate loading step from the main `Player` record so that the page will load more quickly.

Also, it fixes a long-standing annoyance with a hamburger menu rendering at 100% width briefly as pages load.

Additionally, it removes the guild memberships and combines them with the DAO memberships. It increases the number of displayed memberships to ideally be two rows of icons. It also adds a scale animation on hover to make the DAO icons easier to view.

An unrelated improvement is when you click on a user's profile picture it'll copy their ETH address to the clipboard. It's also the hover text for the picture. Otherwise, it's not present on their profile.

**Please provide the GitHub issue number**

Closes #1482, #1478, & #1200.

## Follow up Improvement Ideas

The algorithm for determining how many memberships to display isn't working quite correctly. It is supposed to find the `scrollWidth` of the element & use that to determine how many to display. It's off for some tiles as evidenced by incomplete rows with `+<num>`s at the end.

### CODE REVIEW

**Comment by peth-yursick** [comment]
Looks like the hamburger is still here, just not fullscreen anymore..
image

**Comment by peth-yursick** [comment]
The leaderboard changes are indeed loading blazingly fast tho, which is beautiful to see!

**Comment by dysbulic** [comment]
> Looks like the hamburger is still here, just not fullscreen anymore…

Yeah, the hamburger is the MegaMenu default icon that is hidden once the image loads. I could remove it, but to what end? If for some reason a user didn't load the inverted pyramid image, they'd need that icon.

The real issue is the partially styled page that displays for a second when the page loads, but as best I've been able to determine it's because we're using CSS in JS with Chakra and it has to load everything to be able to know what styles to create.

This would also be helped by my proposal that we begin migrating to vanilla HTML / CSS: #1498, because those styles are generated from the individual components, but from an external stylesheet that should load more quickly.

**Comment by luxumbra** [comment]
I notice that the player avatars aren't displaying. @dysbulic SVG is working but literally every other card is displaying the fallback blockies for the wallet addresses.

image

**Comment by peth-yursick** [comment]
> The real issue is the partially styled page that displays for a second when the page loads, but as best I've been able to determine it's because we're using CSS in JS with Chakra and it has to load everything to be able to know what styles to create.
> 
> This would also be helped by my proposal that we begin migrating to vanilla HTML / CSS: #1498, because those styles are not generated from the individual components, but from an external stylesheet that should load more quickly.

There's plenty of projects using Chakra UI & this is the only place I'm seeing this issue..
Literally anything would be better than having this weird partial list of words or the blown up hamburger, even just a plain black or plain purple page...



> I notice that the player avatars aren't displaying. @dysbulic SVG is working but literally every other card is displaying the fallback blockies for the wallet addresses.

Afaik this has always been the case on test deploys so should be good in prod

**Comment by dysbulic** [comment]
> There's plenty of projects using Chakra UI & this is the only place I'm seeing this issue…

https://user-images.githubusercontent.com/181523/214335448-d25e8fd9-5bab-4636-bc6e-4869cdcb2357.mp4

**Comment by dysbulic** [comment]
> @dysbulic['s] SVG is working but literally every other card is displaying the fallback blockies for the wallet addresses.

I'm looking at the [logs for the test instance backend](https://console.cloud.google.com/run/detail/us-east4/backend-pr-1510/logs?organizationId=295713775631&project=metagame-thegame) and when I view a profile's details I see a cache load on the server, but it's saying there's no profile in Ceramic:

```
No Basic Profile For: 0x2bf034eccebc8cd60dab9c249b6c2996dcb7d8ec (did:3:bafyreicjoorly7mccwukwutspyzzfk3p4pxu2wcip37t5uuf7oy5aqdgzi)
No Extended Profile For: 0x2bf034eccebc8cd60dab9c249b6c2996dcb7d8ec (did:3:bafyreicjoorly7mccwukwutspyzzfk3p4pxu2wcip37t5uuf7oy5aqdgzi)
No Profile Information For 0x2bf034eccebc8cd60dab9c249b6c2996dcb7d8ec.
{
msg: 'Completed Profile Update',
count: 1,
playerId: '0f7d4ff3-d74f-42c8-b727-e4b436138c68',
success: true,
ceramic: 'https://ceramic.metagame.wtf',
did: 'did:3:bafyreicjoorly7mccwukwutspyzzfk3p4pxu2wcip37t5uuf7oy5aqdgzi',
ethereumAddress: '0x2bf034eccebc8cd60dab9c249b6c2996dcb7d8ec',
accountLinks: [],
fields: []
}
```

So, my code is working as expected, but something is amiss with the Ceramic server. I'll debug it in a bit.

**Comment by luxumbra** [comment]
> Learning enough about how to do that in Next.js though is a task for a separate issue.

Fo sho. It would be less intense to just add some lightweight css `<head>` which would add `visibility: hidden` to the container while the page is loading. Been a while since i dabbled with TheGame code but that would be possible no?

**Comment by peth-yursick** [comment]
> You've not seen many sites of the complexity and deeply nested structures that ours is.

Hard to tell which exactly are Chakra but I know eg. DAOhaus is & its probably fairly complex.

Looks pretty irrelevant when looking at it on your recording but I get stuck with that screen for like 3 seconds 😂
I don't mind but its a bit embarrassing when demoing..

> Fo sho. It would be less intense to just add some lightweight css `<head>` which would add `visibility: hidden` to the container while the page is loading. Been a while since i dabbled with TheGame code but that would be possible no?

Yeah tbh whatever hacky solution is ok imo. Even if we just replace the hamburger icon with solid color deep purple..
Idk what the simplest solution would be but I could try asking in DAOhaus if needed

**Comment by HHH-GH** [comment]
Is this helpful?

[Prevent FOUC on Next.js + Chakra UI](https://nibras.co/prevent-fouc-on-next-js-chakra-ui/)

> TL;DR, use @emotion/cache and @emotion/server to prevent initial unstyled content

My read of the issue is that the hamburger and text have basic styling from CSS that is inlined in the static output, and it's not until the rest of the stylesheets are loaded that they get fully styled and hidden.

Would be happy to try it out on a local branch and report back.

**Comment by luxumbra** [comment]
> Is this helpful?
> 
> [Prevent FOUC on Next.js + Chakra UI](https://nibras.co/prevent-fouc-on-next-js-chakra-ui/)
> 
> > TL;DR, use @emotion/cache and @emotion/server to prevent initial unstyled content
> 
> My read of the issue is that the hamburger and text have basic styling from CSS that is inlined in the static output, and it's not until the rest of the stylesheets are loaded that they get fully styled and hidden.
> 
> Would be happy to try it out on a local branch and report back.

That sounds good to me @HHH-GH 👍🏻

**Comment by alalonde** [comment]
> Fo sho. It would be less intense to just add some lightweight css `<head>` which would add `visibility: hidden` to the container while the page is loading. Been a while since i dabbled with TheGame code but that would be possible no?

I am quite confident that there is at least a simple band-aid like this that would suffice, but doesn't have to be part of this PR

**Comment by HHH-GH** [comment]
> > Is this helpful?
> > [Prevent FOUC on Next.js + Chakra UI](https://nibras.co/prevent-fouc-on-next-js-chakra-ui/)
> > > TL;DR, use @emotion/cache and @emotion/server to prevent initial unstyled content
> > 
> > 
> > My read of the issue is that the hamburger and text have basic styling from CSS that is inlined in the static output, and it's not until the rest of the stylesheets are loaded that they get fully styled and hidden.
> > Would be happy to try it out on a local branch and report back.
> 
> That sounds good to me @HHH-GH 👍🏻

I'll have a go. It's going to be a fair few days until I'll have some time to look at it tho

**Comment by dysbulic** [comment]
> Would be less intense to just add some lightweight css `<head>` which would add `visibility: hidden` to the container while the page is loading?

If you look at the rendered HTML, there's actually a couple styles:

```html
<style data-next-hide-fouc="true">
    body {
        display: none
    }
</style>
<noscript data-next-hide-fouc="true">
    <style>
        body {
            display: block
        }
    </style>
</noscript>
```

So, the `body` is already being hidden and revealed programatically.

**Comment by dysbulic** [comment]
So, I just [posted](//discord.com/channels/660863154703695893/1067816578546139296) to [Chakra's Discord](//discord.gg/chakra-ui): 

> I am working on a Next.js v12 application — https://metagame.wtf — using Chakra v2.3.6. We are currently having an issue where partially styled content is displayed on initial load with the final version loading half to a couple seconds later.
>
> I've been looking at the pages generated by `next build` and they include the `style`s marked `data-emotion="css-global"`, but not the component specific styles marked `data-emotion="css"`. Those are only inserted into the DOM when the program runs.
>
> Is this normal? I tried playing around with `@emotion/cache` to no avail. What should I do to get the full styles to render on the server?

**Comment by luxumbra** [comment]
Is the FOUC issue blocking this from getting added to develop?

**Comment by dysbulic** [comment]
> Is the FOUC issue blocking this from getting added to develop?

To my knowledge that's all that's outstanding about this. The base issue of the leaderboard load time has been addressed.

I haven't gotten a response from the Chakra team on the FOUC, and, as mentioned above, the content is theoretically hidden until the styles load by Chakra already. It may be possible to do something more on that front, but I'd just as soon leave it for another PR.

File: packages/utils/package.json [hunk 1]
```diff
@@ -15,13 +15,17 @@
     "test": "jest --passWithNoTests"
   },
   "dependencies": {
+    "@ceramicnetwork/http-client": "^2.15.0",
```
Comment (alalonde) [inline packages/utils/package.json:18]: why the new dependencies?

**Review by alalonde — APPROVED** [review]

File: packages/utils/package.json (hunk 1 above)
Comment (dysbulic) [reply packages/utils/package.json:18]: A library conflict where it says, "Can't assign CeramiApi to CeramicApi, their signatures are different." So, trying to get everything using the same dependencies.
